src/Microsoft.TemplateEngine.Edge/Template/TemplateCreator.cs

```diff
@@ -103,7 +103,35 @@ public TemplateCreator(IEngineEnvironmentSettings environmentSettings)
                 return new TemplateCreationResult(CreationResultStatus.NotFound, templateInfo.Name, LocalizableStrings.TemplateCreator_TemplateCreationResult_Error_CouldNotLoadTemplate);
             }
 
-            string? realName = name ?? fallbackName ?? template.DefaultName;
+            string? realName = name is not null ? name : templateInfo.PreferDefaultName ? templateInfo.DefaultName : null;
+            realName = realName is null ? fallbackName : realName;
```

Comment:
are those lines needed? `realName` will be reassigned in the following code.

Reply:
oops, forgot to change that after, thanks for catching it!